src/app/core/filter/filters/dateFilter.ts

```diff
@@ -1,38 +1,30 @@
-import { Entity } from "../../entity/model/entity";
-import { DateRangeFilterConfigOption } from "../../entity-list/EntityListConfig";
+import { DateFilterBase } from "./date-filter-base";
 import { DateRange } from "@angular/material/datepicker";
-import { calculateDateRange } from "../../basic-datatypes/date/date-range-filter/date-range-filter-panel/date-range-filter-panel.component";
 import moment from "moment";
-import { DataFilter, Filter } from "./filters";
+import { calculateDateRange } from "app/core/basic-datatypes/date/date-range-filter/date-range-filter-panel/date-range-utils";
 import { isValidDate } from "../../../utils/utils";
-import { DateRangeFilterComponent } from "../../basic-datatypes/date/date-range-filter/date-range-filter.component";
+import { Entity } from "../../entity/model/entity";
+import { DataFilter } from "./filters";
 
 /**
  * Represents a filter for date values.
- * The filter can either be one of the predefined options or two manually entered dates.
  */
-export class DateFilter<T extends Entity> extends Filter<T> {
-  override component = DateRangeFilterComponent;
```

Comment:
this broke the date filter functionality. I have reverted some of the changes to fix it